src/macrofier/engine.mjs

```diff
@@ -384,7 +384,7 @@ const promoteAndNameSubSchemas = (obj) => {
     while (more) {
       more = false
       Object.entries(obj.components.schemas).forEach(([key, schema]) => {
-        if (schema.type === "object" && schema.properties) {
+        if ((schema.type === "object" || schema.type === "array") && schema.properties) {
```

Comment:
An array will never have properties, so this should never be true...

Reply:
but in c case we have to create handle for array object too. hence need to know the schema details from the module